nautobot_floor_plan/forms.py

```diff
@@ -83,7 +83,12 @@ class FloorPlanTileForm(NautobotModelForm):
 
     floor_plan = DynamicModelChoiceField(queryset=models.FloorPlan.objects.all())
     rack = DynamicModelChoiceField(
-        queryset=Rack.objects.all(), required=False, query_params={"nautobot_floor_plan_floor_plan": "$floor_plan"}
+        queryset=Rack.objects.all(),
+        required=False,
+        query_params={"nautobot_floor_plan_floor_plan": "$floor_plan", "rack_group": "$rack_group"},
+    )
+    rack_group = DynamicModelChoiceField(
+        queryset=RackGroup.objects.all(), required=False, query_params={"nautobot_floor_plan_floor_plan": "$floor_plan"}
```

Comment:
```suggestion
    rack_group = DynamicModelChoiceField(
        queryset=RackGroup.objects.all(),
        required=False,
        query_params={"nautobot_floor_plan_floor_plan": "$floor_plan", "racks": "$rack"},
```

This filters the Rack Group dropdown to only allow groups that includes the Rack selected (if it has been selected)